test/tests/disclosure-faq.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By, Key } = require('selenium-webdriver');
+const assertAriaControls = require('../util/assertAriaControls');
+const assertAttributeValues = require('../util/assertAttributeValues');
+const assertTabOrder = require('../util/assertTabOrder');
+
+const exampleFile = 'disclosure/disclosure-faq.html';
+
+const ex = {
+  buttonSelector: '#ex1 button',
+  answerSelector: '#ex1 p.desc',
+  buttonSelectors: [
+    '#ex1 dt:nth-of-type(1) button',
+    '#ex1 dt:nth-of-type(2) button',
+    '#ex1 dt:nth-of-type(3) button',
+    '#ex1 dt:nth-of-type(4) button'
+  ],
+  answerSelectors: [
+    '#ex1 dd:nth-of-type(1)',
+    '#ex1 dd:nth-of-type(2)',
+    '#ex1 dd:nth-of-type(3)',
+    '#ex1 dd:nth-of-type(4)'
+  ]
+};
+
+
+const waitAndCheckExpandedTrue = async function (t, element) {
+  return t.context.session.wait(async function () {
+    return (await element.getAttribute('aria-expanded')) === 'true';
+  }, 500);
+};
+
+const waitAndCheckExpandedFalse = async function (t, element) {
+  return t.context.session.wait(async function () {
+    return (await element.getAttribute('aria-expanded')) === 'false';
+  }, 500);
+};
+
+
+// Attributes
+
+ariaTest('"aria-controls" attribute on button', exampleFile, 'button-aria-controls', async (t) => {
+  t.plan(1);
+  await assertAriaControls(t, 'ex1', 'button');
+});
+
+ariaTest('"aria-expanded" attribute on button', exampleFile, 'button-aria-expanded', async (t) => {
+  t.plan(6);
+
+  await assertAttributeValues(t, ex.buttonSelector, 'aria-expanded', 'false');
+
+  let buttons = await t.context.session.findElements(By.css(ex.buttonSelector));
+  for (let button of buttons) {
+    await button.click();
```

Comment:
Huh, I thought having an await in a for..of loop would cause a problem, but the Iteration Protocol handles it automatically. Very readable, very nice.